### PR TITLE
feat: Add download button to /data-sets

### LINKED ISSUE #127: [FE] Add download option for data points
As a user I want to also download each data point individually and also all at once.

### PR DESCRIPTION
**Pull request by Fr3574**
I though this might be a good way of doing it, let me know your thoughts :)
Would it make sense to name the files differently, maybe by ID or title?

Closes #127

### CODE REVIEW

**Comment by Corgam** [comment]
I was wondering, maybe for downloading we should do it in BE? A new endpoint where you provide a filterset, which you always anyways have to prepare when saving a trace or you have it already when getting a saved trace. 

In that case, when FE would change in the future by udk, it would be still supported?

**Comment by thdrbrkw** [comment]
I think the idea behind doing it in FE was because we already have the data there. In the end is an endpoint which takes a traceId or filterset the more generic approach.

**Comment by Fr3574** [comment]
The problem with a backend endpoint is that it would give us 2 routes that both give the same results, creating redundancy. In addition, we would have to provide static files on the backend side, the current variant is much simpler and could also be implemented quickly for another FE.

**Comment by thdrbrkw** [comment]
Okay I see. If we extract our method into a FE-service it's easily reusable. However, if we create the file in BE, later on ZIP-archives could be created of the data. 
Maybe we should just start with FE and if we run into problems move to BE?

File: frontend/src/app/view-datasets/view-datasets.component.ts [hunk 1]
```diff
@@ -63,6 +63,27 @@ export class ViewDatasetsComponent implements OnInit, AfterViewInit {
     });
   }
 
+  // Downloads objects by id, if not specified all queried datapoints
+  download(id?: string) {
+    const a = document.createElement('a');
```
Comment (thdrbrkw) [inline frontend/src/app/view-datasets/view-datasets.component.ts:68]: change name of 'a' into something useful

File: frontend/src/app/view-datasets/view-datasets.component.ts [hunk 2]
```diff
@@ -63,6 +63,27 @@ export class ViewDatasetsComponent implements OnInit, AfterViewInit {
...
+    a.href = URL.createObjectURL(
+      new Blob(
+        [
+          JSON.stringify(
+            id
+              ? this.dataSource.data.find((item) => item._id == id)
+              : this.dataSource.data,
+            null,
+            2
```
Comment (thdrbrkw) [inline frontend/src/app/view-datasets/view-datasets.component.ts:77]: Maybe explain the 2 or use constant with meaningful name?

**Review by thdrbrkw — COMMENTED** [review]
As discussed, can you do this as a generic service? So input is everything strinigifyable and it create the file?

**Review by thdrbrkw — DISMISSED** [review]

**Review by thdrbrkw — APPROVED** [review]

**Pull request merged**